enhancements/proposals/1978-scylladbdatacenter-api/README.md

```diff
@@ -0,0 +1,764 @@
+        cpu: 100m
+        memory: 100Mi
+  topologyLabelSelector:
+    topology.kubernetes.io/region: us-east1
+  nodes: 9
+  racks:
+  - name: a
+    topologyLabelSelector:
+      topology.kubernetes.io/zone: us-east1a
+  - name: b
+    topologyLabelSelector:
+      topology.kubernetes.io/zone: us-east1b
+  - name: c
+    topologyLabelSelector:
+      topology.kubernetes.io/zone: us-east1c
+
+```
+
+### API Spec
+
+Proposed API spec:
+```go
+
+const (
+	AvailableCondition   = "Available"
+	ProgressingCondition = "Progressing"
+	DegradedCondition    = "Degraded"
+)
+
+// ScyllaDBDatacenterSpec defines the desired state of ScyllaDBDatacenter.
+type ScyllaDBDatacenterSpec struct {
+	// metadata controls shared metadata for all pods created based on this spec.
+	// +optional
+	Metadata ObjectTemplateMetadata `json:"metadata,omitempty"`
+
+	// clusterName specifies the name of the scylla cluster.
+	// When joining two DCs, their cluster name must match.
+	ClusterName string `json:"clusterName"`
+
+	// datacenterName specifies the name of the scylla datacenter. Used as datacenter name in GossipingPropertyFileSnitch.
+	// If empty, it's taken from the 'scylladbdatacenter.metadata.name'.
+	DatacenterName string `json:"datacenterName"`
+
+	// scyllaDB holds a specification of ScyllaDB.
+	ScyllaDB ScyllaDB `json:"scyllaDB"`
+
+	// scyllaManagerAgent holds a specification of ScyllaDB Manager Agent.
+	// +optional
+	ScyllaDBManagerAgent *ScyllaDBManagerAgent `json:"scyllaDBManagerAgent,omitempty"`
+
+	// imagePullSecrets is an optional list of references to secrets in the same namespace
+	// used for pulling any images used by this spec.
+	// +optional
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+
+	// dnsPolicy defines how a pod's DNS will be configured.
+	// +optional
+	DNSPolicy *corev1.DNSPolicy `json:"dnsPolicy,omitempty"`
+
+	// dnsDomains specifies a list of DNS domains this cluster is reachable by.
+	// These domains are used when setting up the infrastructure, like certificates.
+	// +optional
+	DNSDomains []string `json:"dnsDomains,omitempty"`
+
+	// forceRedeploymentReason can be used to force a rolling restart of all racks in this DC by providing a unique string.
+	// +optional
+	ForceRedeploymentReason string `json:"forceRedeploymentReason,omitempty"`
+
+	// exposeOptions specifies parameters related to exposing ScyllaCluster backends.
+	// +optional
+	ExposeOptions *ExposeOptions `json:"exposeOptions,omitempty"`
+
+	// nodes specify desired total number of nodes in this datacenter.
+	// This number will be balanced out across all specified racks.
+	// Cannot be used together with `nodes` specified within any rack spec.
+	// +optional
+	Nodes *int32 `json:"nodes"`
+
+	// topologyLabelSelector specifies a label selector which will be used to target nodes at specified topology constraints.
+	// Datacenter topologyLabelSelector is merged with rack topologyLabelSelector and then converted into nodeAffinity
```

Comment:
this is now also in rackTemplate which means 4 fields would have to be merged to produce the affinities which may make it a bit complex

Reply:
Moved to rackTemplate